src/main/java/com/bookshop/controller/AuthController.java

```diff
@@ -0,0 +1,122 @@
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequestMapping("/auth")
+@RequiredArgsConstructor
+@RestController
+public class AuthController {
+    private static final String INCORRECT_DATA =
+            """
+                {
+                    "timestamp": "*time of event*",
+                    "status": "INTERNAL_SERVER_ERROR",
+                    "errors": [
+                        "Incorrect user data"
+                    ]
+                }
+            """;
+
+    private final UserService userService;
+    private final AuthenticationService authenticationService;
+
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "JWT token was generated successfully",
+            content = {@Content(mediaType = "application/json",
+                    examples = {@ExampleObject(value = INCORRECT_DATA)}
+                    )}
+            ),
+        @ApiResponse(responseCode = "500", description = "Internal server error",
+            content = {@Content(mediaType = "application/json",
+                    examples = {@ExampleObject(value = INCORRECT_DATA)}
+                    )}
+            )
+    })
+    @Operation(summary = "login method",
+            description = "returns you a JWT token if authentication was successful")
+    @PostMapping("/login")
+    public UserLoginResponseDto login(
+            @RequestBody @Valid UserLoginRequestDto request) {
+        return authenticationService.authenticate(request);
+    }
+
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "registration was successful"),
+        @ApiResponse(responseCode = "500", description = "Internal server error",
+            content = {@Content(mediaType = "application/json",
+                    examples = {@ExampleObject(value = INCORRECT_DATA)}
+                    )}
+            )
+    })
+    @Operation(summary = "register", description = "register a new user in system")
+    @PostMapping("/register")
+    public UserRegistrationResponseDto register(
+            @RequestBody @Valid UserRegistrationRequestDto request)
+            throws RegistrationException {
+        return userService.register(request);
+    }
+
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "role was set to the user"),
+            @ApiResponse(responseCode = "401",
+                    description = "User should be authenticated to do that operation"),
+            @ApiResponse(responseCode = "403",
+                    description = "Only users with role \"ADMIN\" can do such operation"),
```

Comment:
Agreed, but I tried to describe the situation in which this exception can occur. And it will be if users with roles USER or MANAGER try to use this endpoint. Should I rewrite this description?